### PR TITLE
Feat: Allow specifying environment variables when launching custom entries

### LINKED ISSUE #50: Feature: support for definining environment variables for each config entry
As discussed in rolv-apneseth/lib_game_detector#34, it would be very useful to be able to define env vars for entries in the config, especially for those parsed from Itch (e.g. `SDL_VIDEODRIVER=x11`).

### PR DESCRIPTION
**Pull request by enticedwanderer**
This resolves #50.

All units tests including new ones pass. This is my first contribution on rust so um... Feedback appreciated if I got anything wrong of non-idiomatic. :smile:  Tested end to end on my system with the following config:

```
[[entries]]
title = "The Smoke Room"
path_box_art = "/home/alex/.local/share/itch/_entries/thesmokeroom.png"
path_game_dir = "/home/alex/.local/share/itch/the-smoke-room/"
launch_env = { "SDL_VIDEODRIVER" = "x11" }
```

Modified `TSR.sh` in the game folder to append the contents of the env variable when invoked. Then ran it once with the `launch_env` commented out and once with it uncommented. I got 2 different entries as expected. Once `wayland` another time `x11`.

Also tested with custom `launch_command` and `launch_env` together i.e. both are overridden. This also works. This removes the need for me to have launch_command at all for itch.io games (thanks to your effort on #51).

### CODE REVIEW

**Comment by Rolv-Apneseth** [comment]
Ha, nice. Was just about to start looking at it, shoulda told me you wanted to have a go!

Seems fine to me, if you don't mind a couple nits (e.g. `cargo +nightly fmt`). I'll leave comments.

I'd also like to see an entry using and explaining that new field in the example config in the readme too, if you can.

**Comment by Rolv-Apneseth** [comment]
Looking further into it, formatting is really my only nit. Everything else looks good.

Just to be sure though, could you also add a test case that ensures an empty map (`launch_env = {}` I guess) doesn't cause any problems. Doesn't need to test multiple entries, just one is fine (i.e. no need for `test_case`, just use `test`)

File: README.md [hunk 1]
```diff
@@ -115,6 +115,8 @@ The following sources are currently supported:
 
 - Itch ([itch.io](https://itch.io) app)
   - None of these entries have box art or icons, as they are specified as URLs rather than local files.
+  - All itch.io games are invoked directly through a discovered interpreter script
```
Comment (Rolv-Apneseth) [inline README.md]: This part is technically incorrect, since all of my entries (random couple games I installed) were just pointing directly to binaries with no interpreter specified. Can probably just remove the line.

File: README.md [hunk 2]
```diff
@@ -196,6 +205,9 @@ hide = true
 > [!TIP]
 > To run a script with spaces in the launch command, you will need to use `\\` before any space characters, e.g. `launch_command = "/home/user/GOG\\ Games/Stardew\\ Valley/start.sh"`. Other options can be passed in directly without escaping spaces.
 
+> [!INFO]
+> Note that `launch_env` only takes effect on games that are invoked directly by the launch command (defined either in the config file or discovered). For discovered games, only itch.io games are currently supported. All other discovered games are invoked indirectly and the custom environment variables may not be respected in those cases.
```
Comment (Rolv-Apneseth) [inline README.md]: I think this can go too. I'll have a think on how to let people know it might not work how they expect depending on the launcher.
